src/Compilers/Core/Portable/CommandLine/CommonCommandLineParser.cs

```diff
@@ -451,11 +457,25 @@ internal static string RemoveTrailingSpacesAndDots(string path)
                     continue;
                 }
 
-                if (string.Equals(arg, shared, StringComparison.OrdinalIgnoreCase))
+                if (arg.StartsWith(shared, StringComparison.OrdinalIgnoreCase))
                 {
+                    var hasValue = arg.Length > shared.Length && (arg[shared.Length] == ':' || arg[shared.Length] == '=');
```

Comment:
If you specify the maximum number of substrings to return (`count: 2`), then it will split on the first delimiter, then return the rest of the substring, which should be allowed to contain further delimiters.